src/server/journal/streamer.cc

```diff
@@ -11,53 +11,144 @@
 
 namespace dfly {
 using namespace util;
+using namespace journal;
+using namespace facade;
 
-void JournalStreamer::Start(io::Sink* dest, bool send_lsn) {
-  using namespace journal;
-  write_fb_ = fb2::Fiber("journal_stream", &JournalStreamer::WriterFb, this, dest);
+namespace {
+
+iovec IoVec(io::Bytes src) {
+  return iovec{const_cast<uint8_t*>(src.data()), src.size()};
+}
+
+}  // namespace
+
+JournalStreamer::JournalStreamer(journal::Journal* journal, Context* cntx)
+    : journal_(journal), cntx_(cntx) {
+}
+
+JournalStreamer::~JournalStreamer() {
+}
+
+void JournalStreamer::Start(io::AsyncSink* dest, bool send_lsn) {
+  CHECK(dest_ == nullptr && dest != nullptr);
+  dest_ = dest;
+  // write_fb_ = fb2::Fiber("journal_stream", &JournalStreamer::WriterFb, this, dest);
   journal_cb_id_ =
       journal_->RegisterOnChange([this, send_lsn](const JournalItem& item, bool allow_await) {
         if (!ShouldWrite(item)) {
           return;
         }
 
-        if (item.opcode == Op::NOOP) {
+        if (allow_await) {
+          ThrottleIfNeeded();
           // No record to write, just await if data was written so consumer will read the data.
-          return AwaitIfWritten();
```

Comment:
I do not completely understand the semantics of this.
based on the code I think that  `waker_.await([this]() { return !IsStalled() || IsStopped(); });`  does not wait that everything is completely flushed. Am I wrong?